handlers/producer.go

```diff
@@ -46,7 +84,28 @@ func CreateHandleMessage(conn *memphis.Conn) func(*fiber.Ctx) error {
 				log.Errorf("CreateHandleMessage - handleHeaders: %s", err.Error())
 				return err
 			}
-			err = conn.Produce(stationName, "rest_gateway", message, []memphis.ProducerOpt{memphis.ProducerGenUniqueSuffix()}, []memphis.ProduceOpt{memphis.MsgHeaders(hdrs)})
+			tenantName, username, err := getTenantNameAndUserFromToken(headers["Authorization"])
```

Comment:
not good, for every single message you are parsing the token.
Instead you can add this part to the middleware, that anyway parsing the token and add the extracted fields (like we are doing on the broker)